File: ing.c
#include <stdlib.h>
#include <stdio.h>
#include <string.h>


void lecturaArchivo(void);
void manual (void);
void impresion(char*,int);
char* guardado(char*,char*,int,int);

int main (void){

	char sel;

	printf("[1]Lectura de archivo\n[2]Ingreso manual\n[3]Salir\nOpción:");
	scanf("%c", &sel);

	switch(sel)
		{
			case '1':
				lecturaArchivo();
				break;
			case '2':
				manual();
				break;
			case '3':
					break;
		}
	return 0;
}

void lecturaArchivo(){

	int estado = 0;
	int pos, j = 1;

	FILE* salida;
	FILE* archivo;

	char caracteres[100];
	char *ap = &caracteres[0];

	
	salida = fopen("SalidaIng.txt","w+");
	archivo = fopen("Archivo.txt","rt");

	int n = 10000000;
	printf("%i\n", n);

	fgets(caracteres,n,archivo);

	do{
		for (; *ap!='\0';ap++)
			{
		    	if(*ap=='i' && estado == 0){
		    			pos++;
			    		fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
			    		impresion(ap, estado);
			    		estado = 1;
		    	}else{
		    		if(*ap=='n' && estado == 1){
		    			pos++;
			    		fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
			    		impresion(ap, estado);
			    		estado = 2;

		    		}else{
		    			if(*ap=='g' && estado == 2 && *ap+1 == ' '){
		    				pos++; 
		    				fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
		    				impresion(ap, estado);
		    				
		    			}else{
		    					if(*ap == ' '){
		    						pos++;
		    						fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
		    						impresion(ap, estado);
		    						estado = 0;
		    					} else{
		    						pos++;
		    						fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
				    				impresion(ap, estado);
				    				estado = 0;
		    					}
		    				
		    		
		    			}

		    		}
		    	}

			}

	}while(*ap != '\0');
	
	

	
	fclose(archivo);
	fclose(salida);
		
}

void manual(){

	char texto[10000]={};
	char palabras[10000];
	char *ap = &texto[0];
	char *pal = &palabras[0];
	char palabrasGuardadas[10000];
	int estado = 0;
	int pos = 0;
	int posiciones[10000];
	FILE* salida;

	salida = fopen("SalidaIng.txt","w+");
	printf("Ingrese el texto en el que se buscara la terminación ing: ");
	getchar();
    scanf("%[^\n]",texto); pal = ap;

    for (; *ap!='\0';ap++)
		{
	    	if(*ap=='i' && estado == 0){
	    			pos++;
		    		fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
		    		impresion(ap, estado);
		    		estado = 1;
	    	}else{
	    		if(*ap=='n' && estado == 1){
	    			pos++;
		    		fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
		    		impresion(ap, estado);
		    		estado = 2;

	    		}else{
	    			if(*ap=='g' && estado == 2 && *ap+1 == ' '){
	    				pos++; 
	    				fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
	    				impresion(ap, estado);
	    				
	    			}else{
	    					if(*ap == ' '){
	    						pos++;
	    						posiciones[pos] = pos;
	    						fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
	    						impresion(ap, estado);
	    						estado = 0;
	    					} else{
	    						pos++;
	    						fprintf(salida,"Estado:%i\nLetra: %c\n\n",estado,*ap);
			    				impresion(ap, estado);
			    				estado = 0;
	    					}
	    				
	    		
	    			}

	    		}
	    	}


		}
/*
	for(int i = 0; i<pos; i++){
		if(posiciones[i]!=0){
			printf("%i\n", posiciones[i]);
		}
			
	}	  
	for(;*pal!='\0';pal++){
		fprintf(salida, "%c\n", *pal);
	}
*/
	fclose(salida);
}

void impresion(char* ap, int estado){
	printf("Estado:%i\nLetra: %c\n\n",estado,*ap);

}

File: ing/ing.py
import turtle
import time

def main ():
	sel = input("Selecciona la opcion deseada:\n[1] Lectura de archivo\n[2] Ingreso manual\n[3] Grafo\n[4] Salir\n")

	if sel == 1:
		lecturaArchivo()
	elif sel == 2:
		manual()
	elif sel == 3:
		dibujo()
	else:
		print ("Saliendo...")

def lecturaArchivo():
	i = 0
	estado = 0
	palabras =[]
	caracteres = []
	ap = id(caracteres)
	archivo = open("ArchivoPrueba.txt")
	salida = open("SalidaIng.txt","w+")
	
	caracteres = archivo.read()

	for ap in caracteres:
		if (ap == 'i' or ap == 'I') and estado == 0:
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)
			estado = 1
		elif (ap == 'n' or ap == 'N') and estado == 1:
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)
			estado = 2
		elif (ap == 'g' or ap == 'G') and estado == 2 and caracteres[i+1] == ' ':
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)

		elif (ap == 'g' or ap == 'G') and estado == 2 and caracteres[i+1] !=' ':
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)
			estado = 0

		elif ap == ' ':
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap, estado)
			estado = 0
		else:
				salida.write("Estado:")
				salida.write(str(estado))
				salida.write("\nLetra:")
				salida.write(str(ap))
				salida.write("\n\n")
				impresion(ap,estado)
				estado = 0
	
	with open("ArchivoPrueba.txt") as f:
	    for line in f:
	        words = line.split(" ")
	        for word in words:
	           if 'ing' == word[-3:]:
	              palabras.append(word)

	print "Las palabras que aperecen son:"
	for x in palabras:
			print (x)
			salida.write(str(x))
			salida.write("\n")
	
	pregunta()




def manual():
	texto = []
	palabras=[]
	posiciones=[]
	i = 0;
	ap = id(texto)
	id(palabras)
	estado = 0
	pos = 0
	salida = open("SalidaIng.txt","w+")
	texto = raw_input("Ingresa la cadena de texto en la que se buscara la terminacion ing:")
	copia = open("Copia.txt","w+")
	for j in texto:
		copia.write(str(j))
	copia.close()
	for ap in texto:
		if (ap == 'i' or ap == 'I') and estado == 0:
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)
			estado = 1
		elif (ap == 'n' or ap == 'N') and estado == 1:
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)
			estado = 2
		elif (ap == 'g' or ap == 'G') and estado == 2 and texto[i+1] == ' ':
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)

		elif (ap == 'g' or ap == 'G') and estado == 2 and texto[i+1] !=' ':
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap,estado)
			estado = 0

		elif ap == ' ':
			salida.write("Estado:")
			salida.write(str(estado))
			salida.write("\nLetra:")
			salida.write(str(ap))
			salida.write("\n\n")
			impresion(ap, estado)
			estado = 0
		else:
				salida.write("Estado:")
				salida.write(str(estado))
				salida.write("\nLetra:")
				salida.write(str(ap))
				salida.write("\n\n")
				impresion(ap,estado)
				estado = 0
	ap = id(texto[i+1])
	

	
	with open("Copia.txt") as f:
	    for line in f:
	        words = line.split(" ")
	        for word in words:
	           if 'ing' == word[-3:]:
	              palabras.append(word)

	print "Las palabras que aperecen son:"
	for x in palabras:
			print (x)
			salida.write(str(x))
			salida.write("\n")

	pregunta()

def dibujo():

	print "Una vez que termine de ver el grafo, presione enter en la terminal para salir."

	turtle.showturtle()
	turtle.penup()
	turtle.goto(-250,0)
	turtle.pendown()
	turtle.circle(30)
	turtle.penup()
	turtle.goto(-260,25)
	turtle.write("q0",font=("Arial",20,"normal"))
	turtle.penup()
	turtle.goto(-220,30)
	turtle.pendown()
	turtle.forward(135)
	turtle.dot(8)
	turtle.penup()

	turtle.goto(-50,0)
	turtle.pendown()
	turtle.circle(30)
	turtle.penup()
	turtle.goto(-60,25)
	turtle.write("q1",font=("Arial",20,"normal"))
	turtle.penup()
	turtle.goto(-20,30)
	turtle.pendown()
	turtle.forward(140)
	turtle.dot(8)
	turtle.penup()

	turtle.goto(150,0)
	turtle.pendown()
	turtle.circle(30)
	turtle.penup()
	turtle.goto(150,5)
	turtle.pendown()
	turtle.circle(25)
	turtle.penup()
	turtle.goto(140,25)
	turtle.write("q2",font=("Arial",20,"normal"))
	turtle.penup()
	turtle.goto(150,60)
	turtle.right(270)
	turtle.pendown()
	turtle.circle(60,90)
	turtle.goto(-190,120)
	turtle.circle(60,90)
	turtle.dot(8)

	turtle.penup()
	turtle.goto(-250,0)
	turtle.pendown()
	turtle.dot(8)
	turtle.circle(60,90)
	turtle.goto(-110,-60)
	turtle.circle(60,90)

	turtle.penup()
	turtle.goto(-100,125)
	turtle.write("Diferente de ' ' o de final de cadena",font=("Arial",12,"normal"))
	turtle.goto(-180,-75)
	turtle.write("Diferente de g",font=("Arial",12,"normal"))
	turtle.goto(-160,35)
	turtle.write("n",font=("Arial",12,"normal"))
	turtle.goto(65,35)
	turtle.write("g",font=("Arial",12,"normal"))
	turtle.goto(-390,30)
	turtle.pendown()
	turtle.goto(-280,30)
	turtle.dot(7)
	turtle.penup()
	turtle.goto(-420,30)
	turtle.write("Inicio",font=("Arial",12,"normal"))
	turtle.goto(-370,35)
	turtle.write("Diferente i o i",font=("Arial",12,"normal"))
	raw_input()
	pregunta()


def pregunta():
	opcion = raw_input("Desea regresar al menu\n[1] Si\n[2] No\n")
	if opcion == '2':
		print "Saliendo..."
	else:
		main()
		
		


def impresion(ap,estado):
	print "Estado:",estado,"\nLetra:"+ap,"\n"

main()